code/nebula/neb.cpp

```diff
@@ -191,62 +201,98 @@ void neb2_init()
 
 		// poofs
 		while (required_string_one_of(3, "#end", "+Poof:", "$Name:")) {
-
+			bool create_if_not_found = true;
 			if (Poof_info.size() < MAX_NEB2_POOFS) {
-				poof_info new_poof;
+				poof_info pooft;
+				poof_info* poofp;
+				bool poof_new = true;
 
 				if (optional_string("+Poof:")) { // retail style
 					stuff_string(name, F_NAME, MAX_FILENAME_LEN);
-					strcpy_s(new_poof.bitmap_filename, name);
+					strcpy_s(pooft.bitmap_filename, name);
 
-					strcpy_s(new_poof.name, name);
+					strcpy_s(pooft.name, name);
 
-					generic_anim_init(&new_poof.bitmap, name);
+					generic_anim_init(&pooft.bitmap, name);
 
-					Poof_info.push_back(new_poof);
+					Poof_info.push_back(pooft);
 				} else if (optional_string("$Name:")) { // new style
-					stuff_string(new_poof.name, F_NAME, NAME_LENGTH);
+					stuff_string(pooft.name, F_NAME, NAME_LENGTH);
 
-					required_string("$Bitmap:");
-					stuff_string(name, F_NAME, MAX_FILENAME_LEN);
+					if (optional_string("+nocreate")) {
+						if (!Parsing_modular_table) {
+							Warning(LOCATION, "+nocreate flag used for nebula poof in non-modular table\n");
+						}
+						create_if_not_found = false;
+					}
+
+					// Does this poof exist already?
+					// If so, load this new info into it
+					poofp = get_nebula_pointer(pooft.name);
+					if (poofp != NULL) {
+						if (!Parsing_modular_table) {
+							error_display(1,
+								"Error:  Nebula Poof %s already exists.  All nebula poof names must be unique.",
+								pooft.name);
+						}
+						poof_new = false;
+					} else {
+						// Don't create poof if it has +nocreate and is in a modular table.
+						if (!create_if_not_found && Parsing_modular_table) {
+							if (!skip_to_start_of_string_either("$Name:", "#end")) {
+								error_display(1, "Missing [#end] or [$Name] after nebula poof %s", pooft.name);
+							}
+							continue;
+						}
+						Poof_info.push_back(pooft);
+						poofp = &Poof_info[Poof_info.size() - 1];
+					}
 
-					strcpy_s(new_poof.bitmap_filename, name);
-					generic_anim_init(&new_poof.bitmap, name);
+					if (poof_new) {
+						required_string("$Bitmap:");
+						stuff_string(name, F_NAME, MAX_FILENAME_LEN);
+					} else {
+						optional_string("$Bitmap:");
+						stuff_string(name, F_NAME, MAX_FILENAME_LEN);
+					}
```

Comment:
These blocks appear to be the same; typo?

Reply:
No. Bitmap is required, so if we're making a new poof, require the string. Otherwise, it's optional.